plugins/org.komodo.core/src/org/komodo/repository/search/ObjectSearcher.java

```diff
@@ -0,0 +1,433 @@
+     *
+     * @param type the type to be added
+     * @param alias the alias of the type
+     */
+    public void addFromType(String type, String alias) {
+        ArgCheck.isNotEmpty(type);
+        String failMsg = "From clause data should be added prior to where clause data"; //$NON-NLS-1$
+        ArgCheck.isTrue(whereClauses == null, failMsg);
+        ArgCheck.isTrue(customWhereClause == null, failMsg);
+
+        if (fromTypes == null)
+            fromTypes = new LinkedHashSet<FromType>();
+
+        fromTypes.add(new FromType(type, alias));
+    }
+
+    /**
+     * Add a node type to the From clause
+     *
+     * @param type the type to be added
+     */
+    public void addFromType(String type) {
+        addFromType(type, null);
+    }
+
+    /**
+     * @return set of node types for the From clause
+     */
+    private Set<FromType> getFromTypes() {
+        if (fromTypes == null || fromTypes.isEmpty()) {
+            return Collections.emptySet();
+        }
+
+        return fromTypes;
+    }
+
+    private String checkWhereAlias(String alias) {
+        //
+        // Check that where type alias is valid for set of from types
+        //
+        if (getFromTypes().size() == 1 && StringUtils.isEmpty(alias)) {
+            // Only 1 from type and alias is empty so assume alias of single from type
+            alias = getFromTypes().iterator().next().getAlias();
+        } else {
+            //
+            // More than 1 from type or alias is not empty
+            //
+            ArgCheck.isNotEmpty(alias);
+
+            boolean aliasTypeFound = false;
+            for (FromType fromType : getFromTypes()) {
+                if (fromType.getAlias().equals(alias)) {
+                    aliasTypeFound = true;
+                    break;
+                }
+            }
+
+            ArgCheck.isTrue(aliasTypeFound, "Where clause alias is unknown to from clause"); //$NON-NLS-1$
+        }
+
+        return alias;
+    }
+
+    private SetClause findWhereSetClause(String alias, String property) {
+        for (Clause clause : whereClauses) {
+            if (! (clause instanceof SetClause))
+                continue;
+
+            SetClause setClause = (SetClause) clause;
+            if (setClause.getAlias().equalsIgnoreCase(alias) &&
+                setClause.getProperty().equalsIgnoreCase(property)) {
+                return setClause;
+            }
+        }
+
+        return null;
+    }
+
+    /**
+     * Add an IN sub-clause to the WHERE clause, eg. WHERE alias.property IN (value1, value2, value3)
+     *
+     * @param operator the AND/OR operator preceding the clause. Can be <null> if the first clause
+     * @param alias the alias of the type
+     * @param property the name of the property belonging to the type
+     * @param values the value(s) that the property can be
+     */
+    public void addWhereSetClause(LogicalOperator operator, String alias, String property, String... values) {
```

Comment:
The operator method is used by createStatement() in ObjectSearcher to determine the logical operator between two where clauses, eg. where blah OR blah. Logical operators are attached to the right-hand clause and apply prior to the clause they are attached.